aten/src/ATen/SparseCsrTensorImpl.h

```diff
@@ -57,6 +57,22 @@ struct TORCH_API SparseCsrTensorImpl : public TensorImpl {
     return col_indices_.size(-1);
   }
 
+  int64_t batch_dim() const {
+    return crow_indices_.dim() - 1;
+  }
+
+  int64_t block_dim() const {
+    return (layout_ == kSparseBsr || layout_ == kSparseBsc ? 2 : 0);
+  }
+
+  int64_t sparse_dim() const {
+    return 2;
+  }
```

Comment:
Could that be something other than 2 potentially. What about marking it as a `constexpr` to potentially be able to use it as a template parameter for different implementations that some formats might share.